src/librustdoc/html/static/js/scrape-examples.js

```diff
@@ -79,6 +80,19 @@
 
         // Start with the first example in view
         scrollToLoc(example, locs[0][0]);
+
+        const addSourceLineAnchorsHref = () => {
+            onEachLazy(document.querySelectorAll(".src-line-numbers a"), e => {
+                e.href = linkPage + "#" + e.innerText;
```

Comment:
On small pages, it's fine. However, it's a very bad ideas as soon as you start to have more than 5 functions with examples which comes pretty quickly. Either generate directly the URLs (which would increase the size) or don't change it at all.

Reply:
Big question: is it worth it? It'll increase the page size quite a lot.